src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/subjectaccessrequests/SubjectAccessRequestService.kt

```diff
@@ -0,0 +1,167 @@
+import uk.gov.justice.hmpps.kotlin.sar.HmppsPrisonProbationSubjectAccessRequestService
+import uk.gov.justice.hmpps.kotlin.sar.HmppsSubjectAccessRequestContent
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.time.LocalTime
+
+@Service
+class SubjectAccessRequestService(
+  val objectMapper: ObjectMapper,
+  val cas1SubjectAccessRequestRepository: CAS1SubjectAccessRequestRepository,
+  val cas2SubjectAccessRequestRepository: CAS2SubjectAccessRequestRepository,
+  val cas3SubjectAccessRequestRepository: CAS3SubjectAccessRequestRepository,
+) : HmppsPrisonProbationSubjectAccessRequestService {
+
+  private val log = LoggerFactory.getLogger(this::class.java)
+
+  fun getCAS1Result(crn: String?, nomsNumber: String?, startDate: LocalDateTime?, endDate: LocalDateTime?): String {
+    val approvedPremisesApplicationsJson = cas1SubjectAccessRequestRepository.getApprovedPremisesApplicationsJson(crn, nomsNumber, startDate, endDate)
+    val apApplicationTimelineJson = cas1SubjectAccessRequestRepository.getApprovedPremisesApplicationTimeLineJson(crn, nomsNumber, startDate, endDate)
+    val apAssessmentsJson = cas1SubjectAccessRequestRepository.getApprovedPremisesAssessments(crn, nomsNumber, startDate, endDate)
+    val apAssessmentClarificationNotes = cas1SubjectAccessRequestRepository.getApprovedPremisesAssessmentClarificationNotes(crn, nomsNumber, startDate, endDate)
+
+    val apBookings = cas1SubjectAccessRequestRepository.bookings(crn, nomsNumber, startDate, endDate)
+    val apBookingExtensions = cas1SubjectAccessRequestRepository.bookingExtensions(crn, nomsNumber, startDate, endDate)
+    val apCancellations = cas1SubjectAccessRequestRepository.cancellations(crn, nomsNumber, startDate, endDate)
+    val apBedMoves = cas1SubjectAccessRequestRepository.bedMoves(crn, nomsNumber, startDate, endDate)
+    val domainEvents = cas1SubjectAccessRequestRepository.domainEvents(crn, nomsNumber, startDate, endDate)
+    val domainEventMetaData = cas1SubjectAccessRequestRepository.domainEventMetadata(crn, nomsNumber, startDate, endDate)
+
+    val placementApplications = cas1SubjectAccessRequestRepository.placementApplications(crn, nomsNumber, startDate, endDate)
+    val placementRequests = cas1SubjectAccessRequestRepository.placementRequests(crn, nomsNumber, startDate, endDate)
+    val placementRequirements = cas1SubjectAccessRequestRepository.placementRequirements(crn, nomsNumber, startDate, endDate)
+    val placementRequirementCriteria = cas1SubjectAccessRequestRepository.placementRequirementsCriteria(crn, nomsNumber, startDate, endDate)
+    val offlineApplications = cas1SubjectAccessRequestRepository.offlineApplications(crn, nomsNumber, startDate, endDate)
+    val bookingNotMades = cas1SubjectAccessRequestRepository.bookingNotMades(crn, nomsNumber, startDate, endDate)
+    val appeals = cas1SubjectAccessRequestRepository.appeals(crn, nomsNumber, startDate, endDate)
```

Comment:
very neat! I think all of these vars as JSON strings? In that case can we please suffix all with 'Json' (like the first 3), or none at all for consistency?

Reply:
good spot, i'll go with option 1 since it gives a clue to the nature of the data. will do this in the next pass.

Reply:
in fact i think i need to do this to all 3 CAS results